sidecar/src/service/blocking.rs

```diff
@@ -6,22 +6,92 @@ use super::{
     SidecarInterfaceRequest, SidecarInterfaceResponse,
 };
 use crate::dogstatsd::DogStatsDAction;
-use datadog_ipc::platform::ShmHandle;
+use datadog_ipc::platform::{Channel, ShmHandle};
 use datadog_ipc::transport::blocking::BlockingTransport;
+use std::sync::Mutex;
 use std::{
     borrow::Cow,
     io,
     time::{Duration, Instant},
 };
+use tracing::info;
 
-/// `SidecarTransport` is a type alias for the `BlockingTransport` struct from the `datadog_ipc`
-/// crate. It is used for sending `SidecarInterfaceRequest` and receiving
-/// `SidecarInterfaceResponse`.
+/// `SidecarTransport` is a wrapper around a BlockingTransport struct from the `datadog_ipc` crate
+/// that handles transparent reconnection.
+/// It is used for sending `SidecarInterfaceRequest` and receiving `SidecarInterfaceResponse`.
 ///
 /// This transport is used for communication between different parts of the sidecar service.
 /// It is a blocking transport, meaning that it will block the current thread until the operation is
 /// complete.
-pub type SidecarTransport = BlockingTransport<SidecarInterfaceResponse, SidecarInterfaceRequest>;
+pub struct SidecarTransport {
+    pub inner: Mutex<BlockingTransport<SidecarInterfaceResponse, SidecarInterfaceRequest>>,
+}
+
+impl SidecarTransport {
+    pub fn reconnect<F>(&mut self, factory: F)
```

Comment:
Could you add utests for all those fonctions?

Reply:
Added some tests, but I'm not really happy with them as they are really coupled to the BlockingTransport.

Any advice is welcome!